frb_codegen/src/utils.rs

```diff
@@ -24,3 +26,50 @@ pub fn with_changed_file<F: FnOnce() -> anyhow::Result<()>>(
 
     Ok(fs::write(&path, content_original)?)
 }
+
+pub fn find_all_duplicates<T>(iter: &[T]) -> Vec<T>
+where
+    T: Eq + Hash + Clone,
+{
+    let mut uniq = HashSet::new();
+    iter.iter()
+        .filter(|x| !uniq.insert(*x))
+        .cloned()
+        .collect::<Vec<_>>()
+}
+
+/// check api defined by users, if no duplicates, then generate all symbols (api function name),
+/// including those generated implicitily by frb
+pub fn get_symbols_if_no_duplicates(configs: &[crate::Opts]) -> Result<Vec<String>, anyhow::Error> {
+    let mut explicit_raw_symbols = Vec::new();
+    let mut all_symbols = Vec::new();
+    for config in configs {
+        let raw_ir_file = config.get_ir_file();
+
+        // for checking explicit api duplication
+        explicit_raw_symbols.extend(raw_ir_file.funcs.iter().map(|f| f.name.clone()));
+
+        // for avoiding redundant generation in dart
+        all_symbols.extend(raw_ir_file.get_all_symbols(config));
+    }
+    let duplicates = find_all_duplicates(&explicit_raw_symbols);
+    if !duplicates.is_empty() {
+        let mut duplicated_symbols = duplicates
+            .iter()
+            .map(|s| s.clone() + ",")
+            .collect::<String>();
+        duplicated_symbols.pop();
```

Comment:
map + pop => join

Reply:
> map + pop => join

Fixed in f338868c3b285d4dad1e22d2a17a4c5d57e3f6e6.